source/Halibut.Tests/Timeouts/SendingAndReceivingRequestMessagesTimeoutsFixture.cs

```diff
@@ -0,0 +1,179 @@
+                    // connection is paused we must additionally wait HalibutLimits.TcpClientHeartbeatSendTimeout 
+                    // for that to complete.
+                    addControlMessageTimeout += HalibutLimits.TcpClientHeartbeatSendTimeout;
+                }
+
+                sw.Elapsed.Should().BeCloseTo(HalibutLimits.TcpClientSendTimeout + HalibutLimits.TcpClientSendTimeout + addControlMessageTimeout, TimeSpan.FromSeconds(8),
+                    "We 'should' wait the send timeout amount of time, however when an error occurs writing to the zip (deflate)" +
+                    "stream we also call dispose which again attempts to write to the stream. Thus we wait 2 times the TcpClientSendTimeout.");
+
+                echo.SayHello("A new request can be made on a new unpaused TCP connection");
+            }
+        }
+        
+        [Test]
+        [TestCaseSource(typeof(ServiceConnectionTypesToTest))]
+        public async Task WhenThenNetworkIsPaused_WhileSendingADataStreamAsPartOfARequestMessage_ATcpWriteTimeoutOccurs_and_FurtherRequestsCanBeMade(ServiceConnectionType serviceConnectionType)
+        {
+            using (var clientAndService = await LatestClientAndLatestServiceBuilder
+                       .ForServiceConnectionType(serviceConnectionType)
+                       .WithPortForwarding(out var portForwarderRef)
+                       .WithEchoService()
+                       .Build(CancellationToken))
+            {
+                var echo = clientAndService.CreateClient<IEchoService>();
+                echo.SayHello("Make a request to make sure the connection is running, and ready. Lets not measure SSL setup cost.");
+
+                var echoServiceTheErrorWillHappenOn = clientAndService.CreateClient<IEchoService>(IncreasePollingQueueTimeout());
+
+                
+                var sw = Stopwatch.StartNew();
+                var e = Assert.Throws<HalibutClientException>(() => echoServiceTheErrorWillHappenOn.CountBytes(DataStreamUtil.From(
+                    firstSend: "hello",
+                    andThenRun: portForwarderRef.Value!.PauseExistingConnections,
+                    thenSend: "All done" + Some.RandomAsciiStringOfLength(100*1024*1024)
+                )));
+                AssertExceptionLooksLikeAWriteTimeout(e);
+                sw.Stop();
+                new SerilogLoggerBuilder().Build().Error(e, "Received error when making the request (as expected)");
+                
+                // It is not clear why listening doesn't seem to wait to send a control message here.
+                var addControlMessageTimeout = TimeSpan.Zero;
+                if (serviceConnectionType == ServiceConnectionType.Listening)
+                {
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    {
+                        // It is not clear why windows appears to add this timeout.
```

Comment:
Bug?

Reply:
I am not sure, I didn't investigate this particular timeout. Just noted and moved on :D